### PR TITLE
Introduce assertion methods

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: src/AbstractEnum.php [hunk 1]
```diff
@@ -136,6 +153,21 @@
...
+     * Asserts a constant with this value is defined.
+     *
+     * @param int|string $value  the value to test
+     * @param bool       $strict check the types of the value in the values
+     *
+     * @throws InvalidEnumValue
+     */
+    final public static function assertValidValue($value, $strict = true)
+    {
+        if (!self::isValidValue($value, $strict = true)) {
```
Comment (Soullivaneuh) [inline src/AbstractEnum.php]: I don't think ` = true` was needed here.

File: src/AbstractEnum.php [hunk 2]
```diff
@@ -121,6 +123,21 @@
...
+     * Asserts a constant with this name is defined.
+     *
+     * @param int|string $value  the value to test
+     * @param bool       $strict check the types of the value in the values
+     *
+     * @throws InvalidEnumName
+     */
+    final public static function assertValidName($name)
+    {
+        if (!self::isValidName($name)) {
```
Comment (Soullivaneuh) [inline src/AbstractEnum.php]: `static`? http://php.net/manual/en/language.oop5.late-static-bindings.php

File: src/AbstractEnum.php (hunk 1 above)
Comment (Soullivaneuh) [inline src/AbstractEnum.php]: `static`? http://php.net/manual/en/language.oop5.late-static-bindings.php

File: src/Exception/InvalidEnumName.php [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Greg0ire\Enum\Exception;
+
+final class InvalidEnumName extends \UnexpectedValueException
+{
+    public static function fromName($name, array $supportedNames)
+    {
+        throw new self(sprintf(
```
Comment (Soullivaneuh) [inline src/Exception/InvalidEnumName.php]: `static`? http://php.net/manual/en/language.oop5.late-static-bindings.php

File: src/Exception/InvalidEnumValue.php [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Greg0ire\Enum\Exception;
+
+final class InvalidEnumValue extends \UnexpectedValueException
+{
+    public static function fromValue($value, array $supportedValues)
+    {
+        throw new self(sprintf(
```
Comment (Soullivaneuh) [inline src/Exception/InvalidEnumValue.php]: `static`? http://php.net/manual/en/language.oop5.late-static-bindings.php

File: src/AbstractEnum.php (hunk 2 above)
Comment (greg0ire) [inline src/AbstractEnum.php]: `IsValidName` is final, so, that would be useless, right?

File: src/AbstractEnum.php (hunk 1 above)
Comment (greg0ire) [inline src/AbstractEnum.php]: Oh right, that's even a bug!

Comment (greg0ire) [inline src/AbstractEnum.php]: nein

File: src/Exception/InvalidEnumName.php (hunk 3 above)
Comment (greg0ire) [inline src/Exception/InvalidEnumName.php]: nope

File: src/Exception/InvalidEnumValue.php (hunk 4 above)
Comment (greg0ire) [inline src/Exception/InvalidEnumValue.php]: niet

File: src/AbstractEnum.php (hunk 2 above)
Comment (Soullivaneuh) [inline src/AbstractEnum.php]: In this case maybe, but this change nothing using the keyword which covers every case IMHO. :wink:

**Pull request merged**